eng/Configurations.props

```diff
@@ -9,4 +9,23 @@
     <!-- There's no checked configuration on Mono. -->
     <MonoConfiguration Condition="'$(MonoConfiguration)' == '' and '$(RuntimeConfiguration.ToLower())' == 'checked'">Debug</MonoConfiguration>
   </PropertyGroup>
+
+  <PropertyGroup>
+    <!--
+      Use MSBuild property functions for platform detection:
+      https://docs.microsoft.com/en-us/visualstudio/msbuild/property-functions?view=vs-2019#msbuild-property-functions
+      Note that some of these intrinsic functions are implemented using RuntimeInformation:
+      https://github.com/Microsoft/msbuild/blob/3a9d1d2ae23e41b32a612ea6b0dce531fcf86be7/src/Build/Evaluation/IntrinsicFunctions.cs#L431
+    -->
+    <__BuildOS Condition="$([MSBuild]::IsOSPlatform('OSX'))">OSX</__BuildOS>
+    <__BuildOS Condition="'$(__BuildOS)' == '' and $([MSBuild]::IsOSPlatform('FREEBSD'))">FreeBSD</__BuildOS>
+    <__BuildOS Condition="'$(__BuildOS)' == '' and $([MSBuild]::IsOSPlatform('NETBSD'))">NetBSD</__BuildOS>
+    <__BuildOS Condition="'$(__BuildOS)' == '' and $([MSBuild]::IsOSUnixLike())">Linux</__BuildOS>
+    <__BuildOS Condition="'$(__BuildOS)' == '' and $([MSBuild]::IsOSPlatform('WINDOWS'))">Windows_NT</__BuildOS>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <BuildOS Condition="'$(BuildOS)' == ''">$(__BuildOS)</BuildOS>
```

Comment:
Why do we need the intermediate `__BuildOS` property?

Reply:
It comes form coreclr and they are doing this for every property. I don't see any specific reason for having these properties and could be replaced.
We should do that in a separate PR as we should do it for all other properties like __BuildType, __BuildArch, __LogDir

Reply:
OK. CoreClr is doing this because they are passing in properties from the build scripts and then msbuild modifies these properties in some cases. This will hopefully be fixed with @jkoritzinsky's managed build isolation work.